Sources/Segment/Plugins/SegmentDestination.swift

```diff
@@ -120,20 +133,20 @@ public class SegmentDestination: DestinationPlugin {
             for url in data {
                 analytics.log(message: "Processing Batch:\n\(url.lastPathComponent)")
                 
-                let uploadTask = httpClient.startBatchUpload(writeKey: analytics.configuration.values.writeKey, batch: url) { (result) in
+                let uploadTask = httpClient.startBatchUpload(writeKey: analytics.configuration.values.writeKey, batch: url) { [weak self] (result) in
```

Comment:
I kept the `[weak self]` since it seemed like a good small clean up to have anyway.